invenio_app_rdm/theme/templates/semantic-ui/invenio_app_rdm/header.html

```diff
@@ -7,81 +7,96 @@
-                  <a href="{{ item.url }}">{{ item.text|safe }}</a></li>
-                  {%- endif %}
-              </div>
-            {%- endfor %}
-          </ul>
-        {%- endblock navbar_nav %}
-
-        <div class="right menu">
-          <ul class="no-dots-list">
-            <div class="item right-menu">
-            {%- block navbar_right %}
-              <li>{%- include config.THEME_HEADER_LOGIN_TEMPLATE %}</li>
-            {%- endblock navbar_right %}
-            </div>
-          </ul>
-        </div>
+                {%- endblock navbar_search %}
+              {%- endif %}
+          </li>
+            {%- block navbar_nav %}
+              {%- for item in current_menu.submenu('main').children|sort(attribute='order') if item.visible recursive %}
+                  {%- if item.children %}
+                <li class="navbar-item navbar-options>
+                  <div class="dropdown{{ ' active' if item.active else '' }}">
+                    <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true"
+                      aria-expanded="false" href="{{ item.url }}">{{ item.text|safe }} <b class="caret"></b></a>
+                    <ul class="dropdown-menu">{{ loop(item.children|sort(attribute='order')) }}</ul>
+                  </div>
+                </li>
+                  {%- else %}
+                  <li class="navbar-item navbar-options">
 
-      </div>
+                  <li{{ ' class="navbar-item active"'|safe if item.active and loop.depth == 1 else 'navbar-item' }}>
+                    <a href="{{ item.url }}">{{ item.text|safe }}</a></li>
+                    {%- endif %}
+              {%- endfor %}
+          {%- endblock navbar_nav %}
+          <li class="navbar-item navbar-button">
+            <ul class="rdm-sign-buttons">
+              {%- block navbar_right %}
+                <li class="navbar-item navbar-button">{%- include config.THEME_HEADER_LOGIN_TEMPLATE %}</li>
+              {%- endblock navbar_right %}
+            </ul>
+          </li>
+          
+          <li class="toggle">
+            <input class="menu-btn" type="checkbox" id="menu-btn" />
+            <label class="menu-icon" for="menu-btn"><span class="navicon"></span></label>
+          </li>
+      </ul>
+    </div>
     </nav>
     {%- endblock navbar %}
   </div>
-  {%- block flashmessages %}
-  {%- from "invenio_theme/macros/messages.html" import flashed_messages with context -%}
-  {{ flashed_messages() }}
-  {%- endblock %}
-  {%- block breadcrumbs %}
-  {%- include "invenio_theme/breadcrumbs.html" %}
-  {%- endblock breadcrumbs %}
+{%- block flashmessages %}
+{%- from "invenio_theme/macros/messages.html" import flashed_messages with context -%}
+{{ flashed_messages() }}
+{%- endblock %}
+{%- block breadcrumbs %}
+{%- include "invenio_theme/breadcrumbs.html" %}
+{%- endblock breadcrumbs %}
 </header>
+<script language="javascript">
```

Comment:
Question: What is this needed for?

Reply:
It's needed for expand or collapse the navbar menu when clicking on the hamburger icon, basically it sets the `ul` tag to active and all the elements from that "active" list change the `display` property from `none` to `block`.

Reply:
Ok, I think we need to move it to the ``{% block javascript %}`` template block so that it is evaluated at the end of the page (otherwise it's blocking page rendering). I think ideally, it should be in it's own file like this example from Invenio-I18N:

https://github.com/inveniosoftware/invenio-i18n/blob/master/invenio_i18n/assets/semantic-ui/js/invenio_i18n/app.js

https://github.com/inveniosoftware/invenio-i18n/blob/master/invenio_i18n/webpack.py#L29

Reply:
or perhaps it just has to be added here: https://github.com/inveniosoftware/invenio-theme/blob/master/invenio_theme/assets/semantic-ui/js/invenio_theme/theme.js